### PR TITLE
Fix related to eliminating the dcc Dependencies and unsed tests from the code

### PR DESCRIPTION
**Pull request by Azher2Ali**
In the current PR, we have done refactoring to the code in order to eliminate the icgc.dependency imports. These imports are being replaced with imports from an alternative library, bio.overture.score.core

The reason for this change is that there are no more published versions of the dcc packages available. This means that the code can no longer rely on the org.icgc.dcc

The refactored code imports the function from the bio.overture.score class instead. This ensures that the code will continue to function correctly.

Also, This commit removes several test dependencies related to the org.icgc.dcc libraries:

**dcc-auth-server
dcc-common-core
dcc-metadata-core
dcc-metadata-client
dcc-metadata-server**

These dependencies were previously included for testing purposes.

**Reasoning:**

- Since the dcc dependency has also been removed , these test dependencies are no longer necessary.
- Keeping unused dependencies can clutter the project's pom.xml file and potentially lead to issues during compilation or testing.

Moreover, we have Removed incomplete modern auth rule tests

This commit removes the following test files from the score-test module:

- AbstractStorageIntegrationTest.java
- PartitionedBucketFallbackIntegrationTest.java
- PartitionedBucketIntegrationTest.java
- StorageIntegrationTest.java

As, these tests were identified as incomplete for verifying the functionality of modern authentication rules. They likely did not cover all the necessary scenarios to ensure proper access control based on these rules. Also we have migrated the utils classes which were in libraries and used in the project

### CODE REVIEW

File: score-client/src/main/java/bio/overture/score/client/ssl/DumbHostnameVerifier.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package bio.overture.score.client.ssl;
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.SSLSession;
+
+public class DumbHostnameVerifier implements HostnameVerifier {
+  public DumbHostnameVerifier() {}
```
Comment (joneubank) [inline score-client/src/main/java/bio/overture/score/client/ssl/DumbHostnameVerifier.java:7]: We should find a more descriptive name than `Dumb` for this. `NoTestHostnameVerifier` perhaps?

File: score-client/src/main/java/bio/overture/score/client/ssl/DumbX509TrustManager.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package bio.overture.score.client.ssl;
+
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import javax.net.ssl.X509TrustManager;
+
+public class DumbX509TrustManager implements X509TrustManager {
+  public DumbX509TrustManager() {}
```
Comment (joneubank) [inline score-client/src/main/java/bio/overture/score/client/ssl/DumbX509TrustManager.java:8]: Repeating my comment about not calling these `Dumb`. Lets use names that explain the functionality.

File: score-server/src/main/java/bio/overture/score/server/security/ssl/SSLCertificateValidation.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+      SSLContext sslContext = SSLContext.getInstance("TLS");
+      TrustManager[] trustManagerArray = new TrustManager[] {new DumbX509TrustManager()};
+      sslContext.init((KeyManager[]) null, trustManagerArray, (SecureRandom) null);
+      HttpsURLConnection.setDefaultSSLSocketFactory(sslContext.getSocketFactory());
+      HttpsURLConnection.setDefaultHostnameVerifier(new DumbHostnameVerifier());
+    } catch (NoSuchAlgorithmException | KeyManagementException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
```
Comment (joneubank) [inline score-server/src/main/java/bio/overture/score/server/security/ssl/SSLCertificateValidation.java:23]: Does this mimic the SSL Cert verification being done by DCC core? Looks like we're just trusting everything with our trust manager and verifier performing no tests...

File: score-server/src/main/java/bio/overture/score/server/util/Access.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2016 The Ontario Institute for Cancer Research. All rights reserved.
```
Comment (joneubank) [inline score-server/src/main/java/bio/overture/score/server/util/Access.java:2]: Brand new files can provide a copyright with current year.

File: score-server/src/main/java/bio/overture/score/server/util/Collectors.java
Comment (joneubank) [inline score-server/src/main/java/bio/overture/score/server/util/Collectors.java:1]: Do we want to move these `util` classes into `score-core` instead?

Comment (joneubank) [reply score-server/src/main/java/bio/overture/score/server/util/Collectors.java:1]: - Access
- Collectors
- Joiners
- Splitters
- Streams
- VersionUtils

File: score-server/src/test/java/bio/overture/score/server/service/download/ObjectDownloadServiceTest.java [hunk 5]
```diff
@@ -104,7 +103,7 @@ public void setUpMockService() {
     when(mockService.getEntity(objectId)).thenReturn(metadataEntity);
   }
 
-  @Test(expected = IdNotFoundException.class)
```
Comment (joneubank) [inline score-server/src/test/java/bio/overture/score/server/service/download/ObjectDownloadServiceTest.java:107]: Why did we change this? Does not seem related to the dcc dependency change

File: score-test/src/test/java/bio/overture/score/test/auth/AuthClient.java [hunk 6]
```diff
@@ -45,7 +46,7 @@ public class AuthClient {
 
   @SneakyThrows
   public String createAccessToken() {
-    SSLCertificateValidation.disable();
+    //    SSLCertificateValidation.disable();
```
Comment (joneubank) [inline score-test/src/test/java/bio/overture/score/test/auth/AuthClient.java:49]: I saw you added this SSLCert disable class, why then did we disable it here?

File: score-test/src/test/java/bio/overture/score/test/meta/MetadataClient.java [hunk 7]
```diff
@@ -59,41 +53,42 @@ public Entity findEntity(@NonNull String objectId) {
...
+  // {});
+  //
+  //      results.addAll(page);
+  //      pageNumber++;
+  //    }
+
+  //    // Remove potential duplicates due to inserts on paging:
+  //    // See https://jira.oicr.on.ca/browse/COL-491
+  //    return results.stream().distinct().collect(toImmutableList());
+  //  }
```
Comment (joneubank) [inline score-test/src/test/java/bio/overture/score/test/meta/MetadataClient.java:91]: Why is this function removed? Doesn't seem blocked by any dcc dependencies...

File: score-test/src/test/java/bio/overture/score/test/util/Assertions.java [hunk 8]
```diff
@@ -17,7 +17,7 @@
  */
 package bio.overture.score.test.util;
 
-import static org.assertj.core.api.Assertions.fail;
+// import static org.assertj.core.api.Assertions.fail;
```
Comment (joneubank) [inline score-test/src/test/java/bio/overture/score/test/util/Assertions.java:20]: Why is this removed?

File: score-test/src/test/java/bio/overture/score/test/util/Assertions.java [hunk 9]
```diff
@@ -67,7 +67,7 @@ private static void compareNow(File[] fileArr, HashMap<String, File> map) throws
           String cSum1 = checksum(fileArr[i]);
           String cSum2 = checksum(fComp);
           if (!cSum1.equals(cSum2)) {
-            fail(fileArr[i].getName() + "\t\t" + "different");
+            //            fail(fileArr[i].getName() + "\t\t" + "different");
```
Comment (joneubank) [inline score-test/src/test/java/bio/overture/score/test/util/Assertions.java:70]: If this is a failure test case we should be testing, we can't just remove the fail assertion... It needs to be replaced by something. Can you explain to me whats going on in this file?

**Review by joneubank — COMMENTED** [review]
So in general this looks like we expected, removing some test cases that are no longer relevant to Score in the PartitionBucket tests, and removing DCC dependencies where they are found.

You added the dcc common utils into score server, and i think it may be more useful if those live in the score-core package so that they can be shared across the different score packages.

There is quite a bit of code commented out here that doesn't seem related to the dcc-dependencies at all. Could you add some comments to the PR or respond to mine to explain why these are removed? All these changes should be described in the PR description so that someone reading this request in the future can understand why these pieces of functionality were removed.

Additionally, heres some miscellaneous notes on handling copyright notices on the code files and thoughts on commented code:
- Copyright notices
  - All new code files need the notice pasted in (you've added in some places, not all)
  - Update the year in the notice to the current year for every new notice you add
  - Don't need to update the year in all copyright notices, but you may choose to do this in files you are modifying already in the PR
  - If you want to update every file's copyright notice, we should do that on a separate dedicated PR to reduce noise in this PR
- Commented Code
  - Done for the draft so its understandable that this will be removed eventually, but, theres no reason to comment it out only to remove it later. The git diff shows us the original code that was removed, so we can just remove it.

**Comment by Azher2Ali** [comment]
Closing this PR, as the sync up from the latest develop branch was missed due to which there was conflict between the changes which was actually done. Will be creating a new PR which will hold changes related to dcc dependency only

File: score-client/src/main/java/bio/overture/score/client/ssl/DumbHostnameVerifier.java (hunk 1 above)
Comment (Azher2Ali) [reply score-client/src/main/java/bio/overture/score/client/ssl/DumbHostnameVerifier.java]: Have renamed this to NoTestHostnameVerifier and updated the packages for both score-core and score-server

File: score-server/src/main/java/bio/overture/score/server/security/ssl/SSLCertificateValidation.java (hunk 3 above)
Comment (Azher2Ali) [reply score-server/src/main/java/bio/overture/score/server/security/ssl/SSLCertificateValidation.java]: Yes, the SSL verification mimic the code which is there in DCC Libraries and we have migrated the code.

File: score-server/src/main/java/bio/overture/score/server/util/Access.java (hunk 4 above)
Comment (Azher2Ali) [reply score-server/src/main/java/bio/overture/score/server/util/Access.java]: With the latest PR changes have updated the new files with Copyright Notice

File: score-server/src/main/java/bio/overture/score/server/util/Collectors.java
Comment (Azher2Ali) [reply score-server/src/main/java/bio/overture/score/server/util/Collectors.java:1]: All the util classes were moved to score-core as per the feedback

File: pom.xml [hunk 10]
```diff
@@ -101,17 +101,17 @@ ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF S
...
-        <artifactId>score-server</artifactId>
+        <artifactId>score-fs</artifactId>
         <version>${project.version}</version>
       </dependency>
       <dependency>
         <groupId>${project.groupId}</groupId>
-        <artifactId>score-fs</artifactId>
+        <artifactId>score-client</artifactId>
         <version>${project.version}</version>
       </dependency>
```
Comment (joneubank) [inline pom.xml:116]: I don't know why this order was changed, but since we are updating it can we instead organize these alphabetically?

File: pom.xml
Comment (joneubank) [inline pom.xml:1]: Can we also remove the dcc artifactory repository?

File: score-client/pom.xml [hunk 11]
```diff
@@ -247,9 +240,6 @@ ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF S
     <guava.version>19.0</guava.version>
     <jcommander.version>1.35</jcommander.version>
 
-    <!-- Versions - DCC -->
-    <dcc-common.version>4.0.9</dcc-common.version>
```
Comment (joneubank) [inline score-client/pom.xml:251]: 🙏

**Review by joneubank — CHANGES_REQUESTED** [review]
Great stuff. Only change needed is to completely remove references to the dcc artifactory.

File: pom.xml (hunk 10 above)
Comment (Azher2Ali) [reply pom.xml:116]: Not sure why the order was changed, will organise it alphabetically and update the PR

File: pom.xml
Comment (Azher2Ali) [reply pom.xml:1]: Sure thing!

**Review by joneubank — APPROVED** [review]
All good now! Many thanks!
